src/shared/envvars.pas

```diff
@@ -59,7 +59,9 @@ function SetEnvVar(const VarName: AnsiString; const VarVal: AnsiString): Boolean
   begin
     {$IF DEFINED(WINDOWS)}
       Result := utf2win(UTF8String(SysUtils.GetEnvironmentVariable(WideString('USERNAME'))));
-    {$ELSEIF DEFINED(UNIX)}
+    {$ELSEIF DEFINED(UNIX) AND NOT DEFINED(ANDROID)}
+      // Android is not really Unix. Don't try environment variables there, because it may lead to unexpected behaviour and crashes.
+      // On newer Androids, getenv causes game to crash for some reason.
```

Comment:
Please put a line break to maintain the 100 character limit for line length. (2)